stdlib/private/StdlibCollectionUnittest/CheckCollectionType.swift

```diff
@@ -1254,6 +1260,149 @@ extension TestSuite {
       }
     }
 
+    //===------------------------------------------------------------------===//
+    // last(where:)
+    //===------------------------------------------------------------------===//
+
+    let lastPerformanceTest = FindTest(
+      expected: 3,
+      element: 2020,
+      sequence: [ 1010, 2020, 3030, 2020, 4040 ],
+      expectedLeftoverSequence: [ 3030, 2020, 4040 ])
+  
+    self.test("\(testNamePrefix).last(where:)/semantics") {
+      for test in findLastTests {
+        let c = makeWrappedCollectionWithEquatableElement(test.sequence)
+        let closureLifetimeTracker = LifetimeTracked(0)
+        let found = c.last(where: {
+          _blackHole(closureLifetimeTracker)
+          return $0 == wrapValueIntoEquatable(test.element)
+        })
+        expectEqual(
+          test.expected == nil ? nil : wrapValueIntoEquatable(test.element),
+          found,
+          stackTrace: SourceLocStack().with(test.loc))
+        if let expectedIdentity = test.expected {
+          expectEqual(
+            expectedIdentity, extractValueFromEquatable(found!).identity,
+            "last(where:) should find only the first element matching its predicate")
+        }
+      }
+    }
+
+    self.test("\(testNamePrefix).last(where:)/performance") {
+      let test = lastPerformanceTest
+      let closureLifetimeTracker = LifetimeTracked(0)
+      expectEqual(1, LifetimeTracked.instances)
+      let c = makeCollectionOfEquatable(test.sequence.map(wrapValueIntoEquatable))
+      var closureCounter = 0
+      let found = c.last(where: {
+        (candidate) in
+        _blackHole(closureLifetimeTracker)
+        closureCounter += 1
+        return
+          extractValueFromEquatable(candidate).value == test.element.value
+      })
+      expectEqual(
+        test.expected == nil ? nil : wrapValueIntoEquatable(test.element),
+        found,
+        stackTrace: SourceLocStack().with(test.loc))
+      expectEqual(
+        2,
```

Comment:
It is not immediately clear where this hardcoded `2` is coming from, and it will break if, for whatever reason, the `lastPerformanceTest` changes. Maybe it's worth just getting this value by a very dumb and obviously correct code, like a for loop over the `test.sequence`.

Reply:
Or maybe add another field to `FindTest`.

Reply:
👍 I pulled these out into their own `FindLastTest` type, which let me combine the semantics and performance tests.